src/crystal/system/win32/file.cr

```diff
@@ -26,11 +26,83 @@ module Crystal::System::File
   end
 
   def self.open(filename : String, flags : Int32, perm : ::File::Permissions) : {LibC::Int, Errno}
-    flags |= LibC::O_BINARY | LibC::O_NOINHERIT
+    access, disposition, attributes = self.posix_to_open_opts flags, perm
 
-    fd = LibC._wopen(System.to_wstr(filename), flags, perm)
+    handle = LibC.CreateFileW(
+      System.to_wstr(filename),
+      access,
+      LibC::DEFAULT_SHARE_MODE, # UNIX semantics
+      nil,
+      disposition,
+      attributes,
+      LibC::HANDLE.null
+    )
+
+    if handle == LibC::INVALID_HANDLE_VALUE
+      # Map ERROR_FILE_EXISTS to Errno::EEXIST to avoid changing semantics of other systems
+      return {-1, WinError.value.error_file_exists? ? Errno::EEXIST : Errno.value}
+    end
+
+    fd = LibC._open_osfhandle handle, flags
+
+    if fd == -1
+      return {-1, Errno.value}
+    end
+
+    if flags & LibC::O_BINARY > 0
```

Comment:
`flags` is signed, so `!= 0` is probably better for all those bitmask checks (or even `Int#bits_set?`)

Reply:
Oh nice, thanks! That method is much better.